File: src/InsideNet/InsideNet.Web/Models/PersonModel.cs
using Storage.Entities;

namespace InsideNet.Web.Models
{
    public class PersonModel : GuidIdentifiable
    {
        public string Login { get; set; }

        public string Password { get; set; }

        public string FullName { get; set; }

        public PositionModel Position { get; set; }

        public bool IsNewbie { get; set; }

        public string Telegram { get; set; }

        public string Slack { get; set; }

        public string SlackId { get; set; }

        public string Email { get; set; }

        public string PhoneNumber { get; set; }

        public string AvatarUrl { get; set; }
    }
}

File: Web/src/Api/Models/personModel.ts
import { PositionModel } from "./positionModel";
import { DepartmentModel } from "./departmentModel";

export interface PersonModel {
    readonly id?: string;
    readonly login: string;
    readonly password: string;
    readonly fullName: string;
    readonly authScope: string;
    readonly position: PositionModel; //TODO: who changes positions? admin?
    readonly isNewbie: boolean;
    readonly email: string;
    readonly phoneNumber?: string;
    readonly telegram?: string;
    readonly slack?: string;
    readonly slackId: string; //fixme make it required

    // Other (add them...)
    readonly avatarUrl?: string; // url to avatar
    readonly department: DepartmentModel; //TODO: who changes positions? admin? + add to backend
}
